src/actions.js

```diff
@@ -0,0 +1,82 @@
+export const toLogIn = () => {
+  return {
+    type: 'TO_LOG_IN'
+  }
+}
+
+export const toSignUp = () => {
+  return {
+    type: 'TO_SIGN_UP'
+  }
+}
+
+export const toProfiles = () => {
+  return {
+    type: 'TO_PROFILES'
+  }
+}
+
+export const setAUser = (UID) => {
+  return {
+    type: 'SET_USER',
+    UID
+  }
+}
+
+export const firstName = (e) => {
+  return {
+    type: 'ACCEPT_FIRST_NAME',
+    entry: e.target.value
```

Comment:
I'm not crazy about having DOM-related junk like `e.target.value` leak into action code. I think actions should be more "pure". Therefore, it seems to me that these functions (if you keep them) should simply take an `entry` argument, and when they're called, they pull the appropriate value out of the event.

Reply:
fixed